imap-proto/src/parser/rfc3501.rs

```diff
@@ -676,11 +708,48 @@ mod tests {
         }
     }
 
-      #[test]
+    #[test]
     fn test_addresses() {
         match ::parser::rfc3501::address(b"(\"John Klensin\" NIL \"KLENSIN\" \"MIT.EDU\") ") {
             Ok((_, _address)) => {},
             rsp @ _ => panic!("unexpected response {:?}", rsp)
         }
     }
+
+    #[test]
+    fn test_mailbox_data_recent() {
+        match ::parser::rfc3501::mailbox_data_recent(b"5 RECENT\r\n") {
+            Ok((_, Response::MailboxData(MailboxDatum::Recent(num)))) => assert_eq!(5, num),
+            rsp @ _ => panic!("unexpected response {:?}", rsp)
+        }
+    }
+
+    #[test]
+    fn test_capability_data() {
+        // Minimal capabilities
+        match ::parser::rfc3501::capability_data(b"CAPABILITY IMAP4rev1\r\n") {
+            Ok((_, Response::Capabilities(capabilities))) =>
+                assert_eq!(vec!(Capability::IMAP4rev1), capabilities),
+            rsp @ _ => panic!("unexpected response {:?}", rsp)
+        }
+
+        match ::parser::rfc3501::capability_data(b"CAPABILITY XPIG-LATIN IMAP4rev1 STARTTLS AUTH=GSSAPI\r\n") {
+            Ok((_, Response::Capabilities(capabilities))) =>
+                assert_eq!(vec!(Capability::Atom("XPIG-LATIN"), Capability::IMAP4rev1, Capability::Atom("STARTTLS"), Capability::Auth("GSSAPI")), capabilities),
+            rsp @ _ => panic!("unexpected response {:?}", rsp)
+        }
+
+        match ::parser::rfc3501::capability_data(b"CAPABILITY IMAP4rev1 AUTH=GSSAPI AUTH=PLAIN\r\n") {
```

Comment:
Since we're already in `::parser::rfc3501`, please replace all the instances of that with `super` (or just import `super::*` in the tests module).